buildkite/scripts/lagrange-cache-manager.sh

```diff
@@ -0,0 +1,99 @@
+#!/bin/bash
+set -eo pipefail
+
+# Function to create cache directory with subdirectories for pallas and vesta
+create_cache_dir() {
+  local cache_dir=$1
+  mkdir -p "$cache_dir/pallas" "$cache_dir/vesta"
```

Comment:
Suggest moving this line to `restore_cache` function (then `create_cache_dir` can be rmeoved)